crates/ruff_python_ast/src/scope.rs

```diff
@@ -378,12 +445,12 @@ pub enum BindingKind<'a> {
     Builtin,
     ClassDefinition,
     FunctionDefinition,
-    Export(Vec<String>),
+    Export(Export),
     FutureImportation,
-    StarImportation(Option<usize>, Option<String>),
-    Importation(&'a str, &'a str),
-    FromImportation(&'a str, String),
-    SubmoduleImportation(&'a str, &'a str),
+    StarImportation(StarImportation),
+    Importation(Importation<'a>),
+    FromImportation(FromImportation<'a>),
+    SubmoduleImportation(SubmoduleImportation<'a>),
```

Comment:
Learning opportunity for me: when would we want to `Box` these?

Reply:
When we want the enum to have the size of a box instead of the biggest struct, I think.

Reply:
Boxing a variant reduces the size of that variant to a single pointer instead of the size of the inner struct + padding.

I recommend boxing when one variant is significantly larger than all other variants.